helper-cli/src/main/kotlin/commands/packageconfig/CreateCommand.kt

```diff
@@ -82,15 +82,16 @@ internal class CreateCommand : CliktCommand(
         outputDir.safeMkdirs()
 
         val scanResultsStorage = FileBasedStorage(LocalFileStorage(scanResultsStorageDir))
-        val scanResults = scanResultsStorage.read(packageId).getOrDefault(emptyList())
-
-        scanResults.find { it.provenance is RepositoryProvenance }?.let {
-            createPackageConfiguration(it.provenance)
-        }?.writeToFile()
+        val scanResults = scanResultsStorage.read(packageId).getOrDefault(emptyList()).run {
+            listOfNotNull(
+                find { it.provenance is RepositoryProvenance },
+                find { it.provenance is ArtifactProvenance }
```

Comment:
Can't you simply filter for `KnownProvenance`?

Reply:
The current logic ensures there is at most one RepositoryProvenance and one ArtifactProvenance.
Filtering for KnownProvenance has different semantics IIUC.

Reply:
> The current logic ensures there is at most one RepositoryProvenance and one ArtifactProvenance.

I don't think it does: `find` will get you the *first* of either provenance type, if any, but there could be multiple.

Reply:
`find { it.provenance is RepositoryProvenance }` will return the first provenance which is an instance of `RepositoryProvenance` or null. The second find call will do the same for `ArtifactProvenance`. 
So constraints of the built list are:

- contains at most 2 items
- contains at most one instance of `ArtifactProvenance`
- contains at most one instance of `RepositoryProvenance`

If one uses filter by known provenance instead, above contraints will get eliminated.
With above constraints: No file can be written twice as `ArtifactProvenance` goes to `source-artifact.yml` and `RepositoryProvenance` to "vcs.yml".

Reply:
> The current logic ensures there is at most one RepositoryProvenance and one ArtifactProvenance.

Ah, I misunderstood what "there is" refers to: You refer to the output list, whereas I thought you'd be referring to the input list.

All good now!